x/vm/store/snapshotkv/store.go

```diff
@@ -5,24 +5,23 @@ import (
 
 	"github.com/cosmos/evm/x/vm/store/types"
 
-	"cosmossdk.io/store/cachekv"
 	storetypes "cosmossdk.io/store/types"
 )
 
 // Store manages a stack of nested cache store to
 // support the evm `StateDB`'s `Snapshot` and `RevertToSnapshot` methods.
 type Store struct {
 	// Store of the initial state before transaction execution
-	initialStore storetypes.CacheKVStore
+	initialStore storetypes.CacheWrap
 
 	// Stack of cached store
-	cacheStores []storetypes.CacheKVStore
+	cacheStores []storetypes.CacheWrap
```

Comment:
actually I felt a bit hacky like https://github.com/cosmos/evm/pull/587/commits/04f5ae28058294cee9cb968b472b0b794bc6d1df when doing cast for snapshotKVStore